article/app/controllers/ArticleController.scala

```diff
@@ -238,8 +238,19 @@ class ArticleController extends Controller with RendersItemResponse with Logging
           range
         )
       } getOrElse None
-    liveBlogPageModel match {
-      case Some(pageModel) =>
+    (liveBlogPageModel, range) match {
+      case (Some(pageModel), SinceBlockId(_)) =>
+        val cacheTime =
+          if (liveBlog.fields.isLive && pageModel.currentPage.blocks.isEmpty)
+            liveBlog.metadata.cacheTime
+          else
+            CacheTime.NotRecentlyUpdated
+        val liveBlogCache = liveBlog.copy(
+          content = liveBlog.content.copy(
+            metadata = liveBlog.content.metadata.copy(
+              cacheTime = cacheTime)))
+        Left(LiveBlogPage(liveBlogCache, pageModel, StoryPackages(liveBlog, response)))
+      case (Some(pageModel), _) =>
```

Comment:
Looks like `liveBlogCache` definition is duplicated below. Can we only to match against the `blockRange` when defining cacheTime and keep the `liveBlogCache` in one place.

Also can we abstract away the `cacheTime` logic because it is getting quite involve at this point